src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodToStateMachineRewriter.cs

```diff
@@ -116,6 +116,22 @@ internal void GenerateMoveNext(BoundStatement body, MethodSymbol moveNextMethod)
         {
             F.CurrentMethod = moveNextMethod;
 
+            BoundStatement newBody;
+            try
+            {
+                newBody = GenerateMoveNextBody(body);
+            }
+            catch (SyntheticBoundNodeFactory.MissingPredefinedMember ex)
+            {
+                F.Diagnostics.Add(ex.Diagnostic);
+                newBody = new BoundBadStatement(body.Syntax, ImmutableArray.Create<BoundNode>(body), hasErrors: true);
+            }
+
+            F.CloseMethod(newBody);
```

Comment:
I'm surprised by this. It seems we should be able to check the members we'll need much sooner.

I had done a push in that direction (see PR https://github.com/dotnet/roslyn/pull/9900) for async rewriting by checking the well-known members we'll need are available. Could we do something similar here, and check for `RuntimeBinder.CSharpArgumentInfo.Create` sooner?